### PR TITLE
boards: nucleo_wba: Add flash partitions

### PR DESCRIPTION
**Pull request by erwango**
Add flash partitions which can be used with mcuboot.

### CODE REVIEW

File: boards/arm/nucleo_wba52cg/nucleo_wba52cg.dts [hunk 1]
```diff
@@ -138,3 +139,32 @@ stm32_lp_tick_source: &lptim1 {
...
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(64)>;
+		};
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 DT_SIZE_K(416)>;
```
Comment (gautierg-st) [inline boards/arm/nucleo_wba52cg/nucleo_wba52cg.dts:155]: Is there a reason why 416 and not 448 like slot1? Both spaces seem to be the same size (0x70000)

File: boards/arm/nucleo_wba55cg/nucleo_wba55cg.dts [hunk 2]
```diff
@@ -141,11 +142,25 @@ stm32_lp_tick_source: &lptim1 {
...
 
-		/* Last 16K of flash: Min 2 sectors */
-		storage_partition: partition@f8000 {
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(64)>;
+		};
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 DT_SIZE_K(416)>;
```
Comment (gautierg-st) [inline boards/arm/nucleo_wba55cg/nucleo_wba55cg.dts:151]: Same question

File: boards/arm/nucleo_wba52cg/nucleo_wba52cg.dts (hunk 1 above)
Comment (erwango) [reply boards/arm/nucleo_wba52cg/nucleo_wba52cg.dts:155]: No reason. Good catch, will fix.

**Review by gautierg-st — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]
Needs converting to hwmv2

File: boards/st/nucleo_wba55cg/nucleo_wba55cg.dts [hunk 3]
```diff
@@ -141,11 +142,25 @@ stm32_lp_tick_source: &lptim1 {
...
+			reg = <0x00010000 DT_SIZE_K(448)>;
+		};
+		slot1_partition: partition@80000 {
+			label = "image-1";
+			reg = <0x00080000 DT_SIZE_K(448)>;
+		};
+		scratch_partition: partition@f0000 {
+			label = "image-scratch";
+			reg = <0x000f0000 DT_SIZE_K(16)>;
+		};
```
[the remark this answers is not in the document]
Comment (erwango) [reply boards/st/nucleo_wba55cg/nucleo_wba55cg.dts:160]: ok, thanks. I think we should do this widely on all STM32 targets at once. 
From what I see `MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH` is actually the default, so these are not in use anyway. I'll trigger the change.

**Review by gautierg-st — APPROVED** [review]

**Review by ajarmouni-st — APPROVED** [review]

**Pull request merged**